bridge/opentracing/bridge.go

```diff
@@ -497,8 +498,9 @@ func otTagsToOTelAttributesKindAndError(tags map[string]interface{}) ([]attribut
 	for k, v := range tags {
 		switch k {
 		case string(otext.SpanKind):
-			if s, ok := v.(string); ok {
-				switch strings.ToLower(s) {
+			refValue := reflect.ValueOf(v)
+			if refValue.Kind() == reflect.String {
+				switch strings.ToLower(refValue.String()) {
```

Comment:
The need to use this reflect here seemed weird. So I tried to apply your test change, but not this one. And I see the tests still pass, which makes sense, since `ext.SpanKindRPCClient.Value` is a [SpanKindEnum](https://pkg.go.dev/github.com/opentracing/opentracing-go/ext#SpanKindEnum) and therefore, already a string.

There may be an issue with there, but it doesn't seem to be fixed by this change.

Reply:
since the  ext.SpanKindRPCClient.Value is a ext.SpanKindEnum, so if you try to do the type assertion using string it will fail as its expecting ext.SpanKindEnum.

https://gotipplay.golang.org/p/3yC9R-kp2vj

Reply:
I used reflect for the backward compatibility as someone might be using "string" in Tag.value instead of sub type.

Reply:
Could you then add tests making sure those cases are covered?

Reply:
Tests are already there for both cases. 
[SpanKindEnum](https://github.com/open-telemetry/opentelemetry-go/blob/fde4a75be1a087610752dfa946547b3100eb5fb2/bridge/opentracing/mix_test.go#L641)
[string](https://github.com/open-telemetry/opentelemetry-go/blob/fde4a75be1a087610752dfa946547b3100eb5fb2/bridge/opentracing/mix_test.go#L649)

Reply:
Yes. But they still pass even without your runtime change. So they're not actually testing your fix.

Reply:
[SpanKindEnum](https://github.com/open-telemetry/opentelemetry-go/blob/fde4a75be1a087610752dfa946547b3100eb5fb2/bridge/opentracing/mix_test.go#L641) This wil fail without my changes, just tested with existing code.

Reply:
You are absolutely right, my apologies. Yes, the test case therefore seems good.